source/common/network/io_socket_error_impl.cc

```diff
@@ -12,33 +12,26 @@ Api::IoError::IoErrorCode IoSocketError::getErrorCode() const { return error_cod
 
 std::string IoSocketError::getErrorDetails() const { return errorDetails(errno_); }
 
-IoSocketError* IoSocketError::getIoSocketInvalidAddressInstance() {
-  static auto* instance =
-      new IoSocketError(SOCKET_ERROR_NOT_SUP, Api::IoError::IoErrorCode::NoSupport);
-  return instance;
+Api::IoErrorPtr IoSocketError::getIoSocketInvalidAddressError() {
+  return Api::IoError::wrap(
+      new IoSocketError(SOCKET_ERROR_NOT_SUP, Api::IoError::IoErrorCode::NoSupport));
 }
 
-IoSocketError* IoSocketError::getIoSocketEbadfInstance() {
-  static auto* instance =
-      new IoSocketError(SOCKET_ERROR_BADF, Api::IoError::IoErrorCode::NoSupport);
-  return instance;
+Api::IoErrorPtr IoSocketError::create(int sys_errno) {
+  return Api::IoError::wrap(new IoSocketError(sys_errno));
 }
 
-IoSocketError* IoSocketError::getIoSocketEagainInstance() {
-  static auto* instance = new IoSocketError(SOCKET_ERROR_AGAIN, Api::IoError::IoErrorCode::Again);
-  return instance;
+Api::IoErrorPtr IoSocketError::getIoSocketEbadfError() {
+  return Api::IoError::wrap(new IoSocketError(SOCKET_ERROR_BADF, Api::IoError::IoErrorCode::BadFd));
```

Comment:
same question as above

Reply:
Same answer. (Also one of them was returning a singleton but was *not* excluded from the unique_ptr deletion, so would have actually been a crash bug if the same error had occurred twice, in the original ill-considered copy-paste version.)